woocommerce/woocommerce-android
# Issue/4934 adapt search to new filters feature

Work in progress: #4934 

### Description
Multiple small changes in this PR:

 - Removed order status filters from "Search " feature
 - Fix UI glitches: 
     - Hide filters card when search clicked
     - Remove Toolbar elevation from OrderFiltersFragment
     - Add generic empty case for filtered list with no matching results
- Simplify `OrderListViewmodel` and `OrderListFragment` to avoid unnecessary requests to the API or Db
- Removed some unit tests from OrderListViewModelTest that don't make sense anymore, as we do not have 2 tabs with different order lists and we do not have order status filter integrated in the "search" feature either. 
- Remove feature flag. Once this is merged the feature will be active

Pending: 
- Custom date range filters done here: #5260
- Update filters button in Order list and Product list to latest design which is out of scope for this project and will be handled in another GH issue: 

![Screenshot 2021-11-15 at 19 33 39](https://user-images.githubusercontent.com/2663464/141835685-8fcf0a02-b2df-447e-873a-c0a35deb0dac.png)

Other than that the feature is finished and ready to be fully tested. 

### Testing instructions
Play with filtering the order list. Some ideas:

- Apply some filters and kill the app
- Clearing filters and navigate back, checking the different options after dialog appears. 
- Changing selected site then go back and check filters are not lost
- Play with search feature and see everything works as expected

### Images/gif


https://user-images.githubusercontent.com/2663464/141836161-29733dc1-9bfb-45b2-acab-e9771f5e2470.mp4
